pkg/nfd-topology-updater/nfd-topology-updater_test.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nfdtopologyupdater
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha2"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestTopologyUpdater(t *testing.T) {
+
+	Convey("Given a list of Attributes", t, func() {
+
+		attr_two := v1alpha2.AttributeInfo{
+			Name:  "attr_two_name",
+			Value: "attr_two_value",
+		}
+
+		attrList := v1alpha2.AttributeList{
+			v1alpha2.AttributeInfo{
+				Name:  "attr_one_name",
+				Value: "attr_one_value",
+			},
+			attr_two,
+			v1alpha2.AttributeInfo{
+				Name:  "attr_three_name",
+				Value: "attr_three_value",
+			},
+		}
+		attrListLen := len(attrList)
+		attrNames := getListOfNames(attrList)
+
+		Convey("When an existing attribute is updated", func() {
+
+			existingAttribute := v1alpha2.AttributeInfo{
```

Comment:
nit: this looks more like `updatedAttribute` (I'd not resubmit just for this)

Reply:
Done while squashing the fixing commit